Knossos.NET/Models/Mod.cs

```diff
@@ -348,127 +348,221 @@ public List<string> GetModFlagList(bool overrideSettings = false)
-                                            revisions.Add(d);
-                                        }
-                                        else
-                                        {
-                                            equal.Add(d);
-                                        }
-                                    }
+                                    lower.Remove(l);
                                 }
-                            }
-                            //Equal determine if equalOrhigher or revisions are included here, if so remove them
-                            foreach (var eq in equal)
+                            });
+                            //Remove any "higher" that can be sastified by this equal
+                            higher.ToList().ForEach(h =>
                             {
-                                equalOrHigher.ForEach(eqOrHigh =>
+                                if (SemanticVersion.SastifiesDependency(h.version, eq.version))
                                 {
-                                    if (SemanticVersion.SastifiesDependency(eqOrHigh.version, eq.version))
-                                    {
-                                        temp.Remove(eqOrHigh);
-                                    }
-                                });
-                                revisions.ForEach(revs =>
+                                    higher.Remove(h);
+                                }
+                            });
+                        });
+
+                        //~
+                        rev.ToList().ForEach(re => {
+                            //Remove any "rev" that can be sastified by this other rev
+                            rev.ToList().ForEach(r =>
+                            {
+                                if (re != r && SemanticVersion.SastifiesDependency(r.version, re.version!.Replace("~", "")))
                                 {
-                                    if (SemanticVersion.SastifiesDependency(revs.version, eq.version))
-                                    {
-                                        temp.Remove(revs);
-                                    }
-                                });
-                            }
-                            //Revisions determine if >= or other revisions are included in them, if so remove them
-                            foreach (var revs in revisions)
+                                    rev.Remove(r);
+                                }
+                            });
+                        });
+                        rev.ForEach(re => {
+                            //Remove any "lower" that can be sastified by this rev
+                            lower.ToList().ForEach(l =>
                             {
-                                equalOrHigher.ForEach(eqOrHigh =>
+                                if (SemanticVersion.SastifiesDependency(l.version, re.version!.Replace("~", "")))
```

Comment:
Does this properly filter out `l` if `l` is `<=1.0.1` and `re` is `~1.0.2` ?

Same for higher dependencies, would `h` be filtered out when `h` is `>=1.0.2` and `re` is `1.0.1`?

Reply:
On the first one no, that would be a conflict, "<=1.0.1" cant be sastified with "1.0.2" and "~1.0.2" cant be sastified by "1.0.1". So they both will be in

On the second one mmm "~1.0.1" should be sastified by "1.0.2" what would leave >=1.0.2 out thats definately wrong

Reply:
mmm no, i just ran a test (i have another branch were i did the changes to the UI to test)

![image](https://github.com/KnossosNET/Knossos.NET/assets/2444336/dc2aef33-d3fc-480f-bfb1-ff22b4cf87cd)

And it seem to correctly mark that case as a conflict